### PR TITLE
[stories] implement stories CLI

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

This is the first of several PRs which will break up RFC #13637 into smaller chunks

The CLI is a node file watching service that saves and updates `.stories` files which other RN and web clients can interact with.

# How

Refactored the original PR and stripped it down to its core elements - things like the http server were removed as they are not (yet) used and are a bit of unnecessary bloat for the time being. 

# Test Plan

The core functionality of the CLI is to watch any directory at `watchRoot` for files with `.stories` as a file extension. When a file is added or deleted under the `watchRoot`, there should be an update that reflects this change in the `projectRoot` under the `__generated__` folder

Therefore, to test this out, you can add any `.stories` file anywhere in a file system and point the CLI at that directory, and ensure the generated files are working for add and delete operations.

In the future there should be tests added to ensure the formatting and fast refresh are preserved in the generated output, as this is really the most important part of the CLI

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-stories/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Expo Stories
```
Comment (byCedric) [inline packages/expo-stories/README.md:1]: Could be nice to pimp this readme a bit more, e.g. like `packages/html-elements`?

**Review by byCedric — APPROVED** [review]
I didn't test run this, but it looks good to me! Although I'm not sure if we wanted to get rid of `__generated__`? Maybe nicer to get this into `.expo` folder instead?

**Comment by ajsmth** [comment]
> I didn't test run this, but it looks good to me! Although I'm not sure if we wanted to get rid of `__generated__`? Maybe nicer to get this into `.expo` folder instead?

this was what I wanted too but metro does not play well with `.anything` directories :\

File: packages/expo-stories/types/index.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export interface StoryOptions {
```
[the remark this answers is not in the document]
Comment (ajsmth) [reply packages/expo-stories/types/index.ts:1]: I was curious about the nesting in the build folder - TIL - thanks!

File: packages/expo-stories/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "expo-stories",
```
[the remark this answers is not in the document]
Comment (brentvatne) [reply packages/expo-stories/package.json:2]: I believe you can add `expo` as an owner and then we can adjust permissions further within the org

**Review by brentvatne — APPROVED** [review]
rebase / fix prettier errors so check-packages passes then seems lgtm

**Pull request merged**